web-marketplace/src/pages/BuyCredits/BuyCredits.Form.tsx

```diff
@@ -117,7 +121,7 @@ export const BuyCreditsForm = ({
   const [paymentOptionCryptoClicked, setPaymentOptionCryptoClicked] = useAtom(
     paymentOptionCryptoClickedAtom,
   );
-
+  const [currencyCryptoAtom] = useAtom(buyCreditsCryptoCurrencyAtom);
```

Comment:
why do we need this in addition to `data?.[CURRENCY]`?

Reply:
when going to step 2 using USD and then navigate back to step 1, the currency in `data?.[CURRENCY]` is not updated, even when changing the payment type. This change is to keep track of payment type updates so we can fetch the right `userBalance` - otherwise the balance is always 0 in this scenario.

Reply:
I see, this is because `data?.[CURRENCY]` only changes as soon as the user submits step 1
But I find it confusing to have to keep track of this twice in slightly different ways
Maybe we should keep getting the userBalance on the ChooseCreditsForm using `currency` from `useWatch` for this specific case, just like we already instantiate different `filteredCryptoSellOrders` and `orderedSellOrders`